embedded/sql/row_reader.go

```diff
@@ -67,6 +68,31 @@ func (row *Row) Compatible(aRow *Row, selectors []*ColSelector, db, table string
 	return true, nil
 }
 
+func (row *Row) digest() (d [sha256.Size]byte, err error) {
+	h := sha256.New()
+
+	for _, v := range row.Values {
+		_, isNull := v.(*NullValue)
+		if isNull {
+			continue
+		}
```

Comment:
That way we skip null values in the tuples but it comes with some incorrect results, e.g.
`Row(1,null).digest == Row(null,1).digest`.

Can be easily confirmed with the following steps:
```bash
./immuclient exec "CREATE TABLE t1(id INTEGER AUTO_INCREMENT, a INTEGER, b INTEGER, PRIMARY KEY id)"
./immuclient exec "INSERT INTO t1(a,b) VALUES(1,null),(null,1)"
./immuclient query "SELECT DISTINCT a,b FROM t1"
# +------------------+------------------+
# | (DEFAULTDB T1 A) | (DEFAULTDB T1 B) |
# +------------------+------------------+
# |                1 | NULL             |
# +------------------+------------------+
```

For now I'd simply add some null indicator, e.g.:
```suggestion
		_, isNull := v.(*NullValue)
		if isNull {
			h.Write([]byte{0})
			continue
		}
		h.Write([]byte{1})
```

Reply:
actually that may not be enough to prevent overlaps, each encoded value may need to include the offset as prefix